packages/core/src/components/button/button.tsx

```diff
@@ -48,7 +48,9 @@ export class TdsButton {
     }
     return (
       <Host
-        class={`${this.modeVariant !== null ? `tds-mode-variant-${this.modeVariant}` : ''} `}
+        class={{
+          [`tds-mode-variant${this.modeVariant}`]: Boolean(this.modeVariant),
```

Comment:
```suggestion
          [`tds-mode-variant-${this.modeVariant}`]: Boolean(this.modeVariant),
```
The missing dash formats the class name into "tds-mode-variantprimary"